### ISSUE TITLE
confusing GUI while RAID imaging

### ISSUE THREAD

**Issue by GralfR** [rescuezilla/rescuezilla#174]
I've the live-ISO rescuezilla 2.1.2x64 and tried to image-backup my onboard Intel RAID1 with Ubuntu 20.04 LTS on two HDD. The RAID is recognized by the live-ISO as /dev/md126, but rescuezilla does not show this inside the source-GUI for backup - even not with show hidden enabled.
So I've googled and read a lot about imaging RAIDs and was lucky to read, that rescuezilla 2 in some kind of way does support that.

My current task is now to clone both HDD sda and sdb into separate image-files. I hope this works, because it's a RAID1.

Doing so I found it strange inside the GUI: the source-GUI does not list the /dev/md126, only /dev/sda and /dev/sdb. So I selected /dev/sda as a source.

Next thing is to select the partitions. Only one partition of the disk-size is offered by rescuezilla, but I know, the RAID has three partitions. Gparted on the rescuezilla live-ISO shows this. I think this might have to do with a known limitation, because there is also a LVM on the disk?

Well, finally I got to the summary-GUI. This and the next status GUI say it's doing a backup of /dev/md126. That's what I wanted to do, but that's not, what I set up in the wizard before.

So, maybe just the GUI needs to be improved a bit after having thankfully implemented the RAID-core?

...still waiting for my image to be created to know more....

**Comment by shasheene** [comment]
By the way, this kind of support discussion is better done on the [Rescuezilla forum hosted on Sourceforge](https://sourceforge.net/p/rescuezilla/discussion/) (the GitHub issues are for focused tasks with clearer scope).

That separate image workaround is out of date. Since v2.0, Rescuezilla fully supports [software RAID device nodes (eg, /dev/md0) created by the `mdadm` utility](https://www.linuxbabe.com/linux-server/linux-software-raid-1-setup) and hardware RAID, though Rescuezilla still has [other limitations](https://github.com/rescuezilla/rescuezilla/wiki/Rescuezilla-Limitations). Always use the latest version of Rescuezilla as you are already doing.

On my test system I have a /dev/md127 device node. Rescuezilla lists this partition only when the "Show hidden devices (for advanced users)" checkbox is ticked. I was able to make a backup of /dev/md127 without issue. My RAID drive does not have a partition table, during "Step 4: Select Partitions To Restore" the "Overwrite Partition Table" option cannot be ticked (as expected), so I have to use the dropdown menu on the 'Destination partition' column to map the partition. The restore then completes without issues. That was for a drive without a partition table. To double-check the handling of partition tables, I just tried formatting my /dev/md127 device with a partition table and create an NTFS filesystem on it (/dev/md127p1). I was also able to backup and restore without issues.

I wasn't able to reproduce the issue you describe where making a backup of the constituent drive selected the wrong drive. The disk on my test system named /dev/nvme0n1 contains a linux_raid_member filesystem on /dev/nvme0n1p1. I was able to make a backup of that drive without any issues. Carefully making backups of constituent drives is fully support approach in theory, but its easy to make a mistake (or eg, forget to deactivate your /dev/md126) when going the constituent drive route, which could cause problems. Because of this **I definitely suggest avoiding making backups of constituent drives since Rescuezilla supports software RAID**. I'm not expert on RAID. There are so many different ways to configure software RAID using mdadm. But once configured, md software RAID works like any other Linux block device, **and as far as I have tested Rescuezilla supports /dev/mdX nodes**.

Also I should mention, Rescuezilla and Clonezilla follow the same internal behavior as far as possible (and Clonezilla by all accounts has excellent /dev/mdX support) so for some users such as yourself it might be worth comparing Rescuezilla's behavior with Clonezilla.
